### PR TITLE
Milestone 3: Add and remove

### PR DESCRIPTION
**Pull request by medaminedev66**
- Add `task.js` file, create Task class.
- Add `addTask` and `removeTask` functions to `tasks.js` file.
- Add remove icon to `listContainer`.
- Add event listener to add icon and `input-text` class .
- Add event Listener to remove icon.
- Update all remaining items' indexes after deleting a task.

### CODE REVIEW

File: src/tasks.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  if (newList.length > 0) {
+    let i = 1;
+    newList.forEach((element) => {
+      element.index = i;
+      i += 1;
+    });
+  }
+  localStorage.setItem('list', JSON.stringify(newList));
+};
+
```
Comment (henatan99) [inline src/tasks.js:29]: - [ ] In the project's specific requirements, a function for editing task descriptions is required to be implemented. You can add it here or any other file that suits you. Kindly add the edit description method. :pray: You can revisit the project's requirements [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md).
- [ ] In the project's specific requirements, the function for the "Clear all completed" button (use filter() method) is required to be implemented. Again you can revisit the project's requirements [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md).

Comment (henatan99) [inline src/tasks.js:29]: - [ ] In the project's specific requirements, a function for editing task descriptions is required to be implemented. You can add it here or any other file that suits you. Kindly add the edit description method. :pray: You can revisit the project requirements [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md).

File: dist/index.html [hunk 2]
```diff
@@ -1,21 +1 @@
...
-    <title>To do list</title>
-<script defer src="main.js"></script></head>
-
-<body>
-    <section class="to-do-list">
-        <div class="title">
-            <h2>Today's to do</h2>
-        </div>
-        <input type="text" class="input-text" placeholder="...">
-        <div class="the-list"></div>
```
Comment (henatan99) [inline dist/index.html:17]: - [ ] I think you should have a "Clear all completed" button below this element. In this milestone, you are also required to be able to delete all the completed tasks in one button click. :+1: Please look back to the project's requirements [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md).

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @medaminedev66 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: dist/index.html [hunk 3]
```diff
@@ -5,16 +5,21 @@
     <meta charset="UTF-8">
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="https://pro.fontawesome.com/releases/v5.10.0/css/all.css" integrity="sha384-AYmEC3Yw5cVb3ZcuHtOA93w35dYTsvhLPVnYs9eStHfGJvOvKxVfELGroGkvsg+p" crossorigin="anonymous"/>
     <title>To do list</title>
-<script defer src="main.js"></script></head>
+<script defer src="/to-do-list/main.js"></script></head>
```
Comment (hemantso) [inline dist/index.html]: - [ ] On the first instance I tried to run the project locally but functions were not working then I noticed this. It is working perfectly as `<script defer src="main.js"></script>`. Kindly try to update to the correct path.

File: README.md [hunk 4]
```diff
@@ -4,9 +4,14 @@
...
 ## Built With
 
 - Javascript, HTML and CSS
+
+## Live Demo
+
+[Live Demo Link](https://medaminedev66.github.io/to-do-list/)
 ## Getting Started
 
 - In terminal or Command line. Navigate to a directory in your local storage where you'd like to store the project and paste ```https://github.com/medaminedev66/to-do-list.git```
```
Comment (hemantso) [inline README.md]: - [ Optional] Kindly try to update the `Live link` and correct the information for getting started in your README file. Like `npm install`, `npm run build` and `npm start` are some of the information missing to start a project locally.

**Review by hemantso — CHANGES_REQUESTED** [review]
Hi @medaminedev66, 

Good job so far. 👍 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 

Alt Text

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- 

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hemantso — APPROVED** [review]
Hi @medaminedev66, 

Good work on making the requested changes. 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by medaminedev66** [comment]
Thanks to you, @hemant-soni-vst-au4 , for your time to review my project.

**Comment by medaminedev66** [comment]
Thanks to you, @hemant-soni-vst-au4 , for your time to review my project.

**Comment by medaminedev66** [comment]
Thanks to you, @hemant-soni-vst-au4 , for your time to review my project.

**Pull request merged**